examples/api/test/widgets/scroll_view/custom_scroll_view.1_test.dart

```diff
@@ -0,0 +1,244 @@
+        matching: find.byType(Container),
+      ),
+      findsOne,
+      reason: 'Expected one, initial Container in the SliverList',
+    );
+
+    //CustomScrollView does not contain the element 'Item: 1'
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.text('Item: 1'),
+      ),
+      findsNothing,
+      reason: 'Expected no element with text "Item: 1" in the CustomScrollView',
+    );
+
+    // Tap the IconButton in the AppBar
+    final Finder iconButtonFinder = find.descendant(
+      of: find.byType(AppBar),
+      matching: find.byType(IconButton),
+    );
+    await tester.tap(
+      iconButtonFinder,
+    );
+    await tester.pump();
+
+    //Now there are two Containers on visible SliverList
+    expect(
+      find.descendant(
+        of: find.byType(SliverList),
+        matching: find.byType(Container),
+      ),
+      findsExactly(2),
+      reason:
+          'There is no additional Container in the SliverList after the IconButton click',
+    );
+
+    //CustomScrollView now contains the element 'Item: 1'
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.text('Item: 1'),
+      ),
+      findsOne,
+      reason: 'Expected element with text "Item: 1" in the CustomScrollView',
+    );
+  });
+  testWidgets('IconButton click and mouse scroll reveals additional SliverList',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(const example.CustomScrollViewExampleApp());
+
+    //CustomScrollView does not contain the element 'Item: -1'
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.text('Item: -1'),
+      ),
+      findsNothing,
+      reason:
+          'Expected no element with text "Item: -1" in the CustomScrollView',
+    );
+
+    //CustomScrollView does not contain the element 'Item: 1'
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.text('Item: 1'),
+      ),
+      findsNothing,
+      reason: 'Expected no element with text "Item: 1" in the CustomScrollView',
+    );
+
+    // First check before we start
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.byType(SliverList),
+      ),
+      findsOne,
+      reason:
+          'Expected to have only one, initial (bottom), SliverList in the CustomScrollView',
+    );
+
+    // Second check before we start.
+    // Initially, mouse scroll event should do nothing.
+    // It should not reveal additional (top) SliverList because additional SliverList does not exists yet.
+    final Offset location = tester.getCenter(find.byType(CustomScrollView));
+    final TestPointer testPointer = TestPointer(1, PointerDeviceKind.mouse);
+    testPointer.hover(location);
+    await tester.sendEventToBinding(PointerScrollEvent(
+        position: location, scrollDelta: const Offset(0, -1)));
```

Comment:
Not a huge deal whatsoever, but maybe consider making a whitespace change for readability:

```suggestion
    await tester.sendEventToBinding(
      PointerScrollEvent(position: location, scrollDelta: const Offset(0, -1)),
    );
```